connections/sim/hw/pin_sim.py

```diff
@@ -0,0 +1,29 @@
+
+
+class Pin:
+    """
+    provides generic pin attributes to be inherited by other pin classes
+    for HW sim
+    """
+
+    INPUT = 0
+    OUTPUT = 1
+
+    def __init__(self):
+        #could be updated later.  For now, sets the pin_mode to be default as an output
+        self.pin_mode = self.OUTPUT
```

Comment:
Lets make the default be a 3rd state to indicate that the pin has not been set yet. Then we can also use this in the SIM test to make sure that the FW is actually setting the pins.

Reply:
I've added this now, however the implementation is still just a dictionary member in the hw_sim class.  I can adjust the implementation of PinMode in a later PR